dwh/query_lib_compatibility_tests.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from functools import partial
+import base as ql_base
+import query_lib as ql
+
+_BIGQUERY_DATASET = "synthea_big"
+_PROJECT_NAME = "fhir-analytics-test"
+_CODE_SYSTEM = "http://www.ampathkenya.org"
+
+_SPARK_BASE_DIR = "./test_files/parquet_big_db"
+
+
+class _PatientQueryTest:
+    """
+    Base class that holds all actual tests that generic to all query lib implementations.
+    """
+
+    def test_encounter_basic_query(self):
+
+        pq = self._PatientQueryClass()
```

Comment:
Please see comment below regarding using abstract methods vs `partial`. Once that is done, all of these calls at the beginning of each test to create a `PatientQuery` instance, should be moved to the `setUp` function. `_PatientQueryTest` should inherit from `unittest.TestCase` and the other two classes inherit from `_PatientQueryTest` only (and not `unittest.TestCase`).

Reply:
As I commented in the other comment, we cannot reuse the same instance of patient query object since state cannot be shared.

Reply:
I am creating an instance at the beginning of every test.